app/helpers/current_helper.rb

```diff
@@ -0,0 +1,3 @@
+module CurrentHelper
+  thread_mattr_accessor :current_scope_user
```

Comment:
This looks a bit arcane to me - you're creating a thread-global constant? Without being familiar with the code, `CurrentHelper` isn't very intuitive. What problem are you solving with this?

Reply:
It looks like a lot of this was written to try and get the name of the current user that was logged in so that the modification can be logged to the system.  However I think ```Auditable``` takes care of this.  @TheDudeWithTheThing I know you wrote a lot of the audit logic.  My spider sense things that some of the stuff in this PR shouldn't be needed if you can take a look too.

Reply:
I tried to get the name of the current user in Reference model so that the changes can be logged in comment section.